builtin/logical/transit/policy.go

```diff
@@ -335,11 +358,27 @@ func (p *Policy) DeriveKey(context []byte, ver int) ([]byte, error) {
 		return nil, errutil.UserError{Err: "missing 'context' for key deriviation. The key was created using a derived key, which means additional, per-request information must be included in order to encrypt or decrypt information"}
 	}
 
-	switch p.KDFMode {
-	case kdfMode:
+	switch p.KDF {
```

Comment:
By the looks of it, without a policy object DeriveKey cannot be invoked. While fetching the policy itself, the upgrade path must be walked to get the `p.KDF` populated. Just to double confirm it, we don't need a `needsUpgrade` check in this function, right?